shared/transformations/lib/flattenYMap.ts

```diff
@@ -0,0 +1,28 @@
+import { type EleBlockGroup } from '@/shared/types/index.js'
+import { yTextToSlateElement } from '@slate-yjs/core'
+import { Node } from 'slate'
+import * as Y from 'yjs'
+
+type Result = Record<string, EleBlockGroup | string> | Record<string, unknown>
+
+export function flattenYMap(ymap: Y.Map<unknown>): Result {
+  const result: Result = {}
+
+  for (const [key, value] of ymap.entries()) {
+    if (value instanceof Y.Map) {
+      result[key] = flattenYMap(value)
+    } else if (value instanceof Y.Array) {
+      result[key] = value.toArray()
+        .map((item) => item instanceof Y.Map ? flattenYMap(item) : item)
```

Comment:
To get rid of the lint warning:
```suggestion
        .map((item): unknown => item instanceof Y.Map ? flattenYMap(item) : item)
```